### PR TITLE
feat: enable test file crawling for jest example

### PR DESCRIPTION
**Pull request by mrmeku**
## PR Checklist
Before, the jest example macro used the `runTestsByPath` command line flag to specify test locations. Now, using a small patch to jest to enable symlinks, the more iodmatic method of using a test match glob pattern is used.

`testMatch: ['**/*.test.js']`

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,3 @@
 {
```
Comment (alexeagle) [inline .vscode/settings.json]: do you mean to check in this change?

File: examples/jest/patches/jest-haste-map+24.9.0.patch [hunk 2]
```diff
@@ -0,0 +1,16 @@
+diff --git a/node_modules/jest-haste-map/build/crawlers/node.js b/node_modules/jest-haste-map/build/crawlers/node.js
+index 23985ae..a728a9a 100644
+--- a/node_modules/jest-haste-map/build/crawlers/node.js
++++ b/node_modules/jest-haste-map/build/crawlers/node.js
+@@ -166,7 +166,11 @@ function find(roots, extensions, ignore, callback) {
+ 
+ function findNative(roots, extensions, ignore, callback) {
+   const args = Array.from(roots);
++  args.push('(');
```
Comment (alexeagle) [inline examples/jest/patches/jest-haste-map+24.9.0.patch]: is there an issue or PR open on the upstream repo so there's a chance of unwinding this patch later?
if so add a comment somewhere pointing to it, so we'll know when it's time to unpatch

File: examples/jest/jest.bzl [hunk 3]
```diff
@@ -10,8 +10,6 @@ def jest_test(name, srcs, deps, jest_config, **kwargs):
         "--ci",
     ]
     args.extend(["--config", "$(location %s)" % jest_config])
```
Comment (alexeagle) [inline examples/jest/jest.bzl]: I was expecting to see a glob pattern being added here or in the place jest_test is called

File: examples/jest/patches/jest-haste-map+24.9.0.patch [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+   args.push('-type', 'f');
++  args.push('-o');
++  args.push('-type', 'l');
```
Comment (alexeagle) [inline examples/jest/patches/jest-haste-map+24.9.0.patch]: I don't quite understand where the symlinks are appearing. The node-patches should make it so the source files appear to be in the runfiles tree, not symlinks out of the runfiles. Are you sure the fix belongs in jest and not in the node-patches? Can you show where the symlink appears and where it points?

File: .vscode/settings.json (hunk 1 above)
Comment (mrmeku) [reply .vscode/settings.json]: Maybe it shouldn't be part of this change, but in general I have to do this to rules_nodejs everytime I try to edit the repo in vscode. It breaks the bazel vscode extensions ability to format build files

File: examples/jest/jest.bzl (hunk 3 above)
Comment (mrmeku) [reply examples/jest/jest.bzl]: The glob pattern was already present within `jest.config.js` (that's the idiomatic way of specifying tests to jest). Do you have a different API you would prefer?

File: examples/jest/patches/jest-haste-map+24.9.0.patch (hunk 2 above)
Comment (mrmeku) [reply examples/jest/patches/jest-haste-map+24.9.0.patch]: For my own reference, he's the PR I've started with Jest: https://github.com/facebook/jest/pull/9351

File: examples/jest/patches/jest-haste-map+24.9.0.patch (hunk 4 above)
Comment (mrmeku) [reply examples/jest/patches/jest-haste-map+24.9.0.patch]: The symlinks appear within the runfiles tree of the testable target. `nodejs_binary` creates a runfiles directory which is filled with symlinks to the original files.

**Comment by alexeagle** [comment]
In a followup could you add something to the examples/jest README so users understand they also have to copy and apply the patch if they want their globs to work?

**Pull request merged**